test/jasmine/tests/table_test.js

```diff
@@ -0,0 +1,972 @@
+mock0.data[0].dimensions = [];
+
+var mock = require('@mocks/table_large.json');
+
+var lineStart = 30;
+var lineCount = 10;
+
+describe('table initialization tests', function() {
+
+    'use strict';
+
+    describe('table global defaults', function() {
+
+        it('should not coerce trace opacity', function() {
+            var gd = Lib.extendDeep({}, mock1);
+
+            Plots.supplyDefaults(gd);
+
+            expect(gd._fullData[0].opacity).toBeUndefined();
+        });
+
+        it('should use global font as label, tick and range font defaults', function() {
+            var gd = Lib.extendDeep({}, mock1);
+            gd.layout.font = {
+                family: 'Gravitas',
+                size: 20,
+                color: 'blue'
+            };
+
+            Plots.supplyDefaults(gd);
+
+            var expected = {
+                family: 'Gravitas',
+                size: 17,
+                color: 'blue'
+            };
+
+            expect(gd._fullData[0].labelfont).toEqual(expected);
+            expect(gd._fullData[0].tickfont).toEqual(expected);
+            expect(gd._fullData[0].rangefont).toEqual(expected);
+        });
+    });
+
+    describe('table defaults', function() {
+
+        function _supply(traceIn) {
+            var traceOut = { visible: true },
+                defaultColor = '#444',
+                layout = { font: Plots.layoutAttributes.font };
+
+            Table.supplyDefaults(traceIn, traceOut, defaultColor, layout);
+
+            return traceOut;
+        }
+
+        it('\'line\' specification should yield a default color', function() {
+            var fullTrace = _supply({});
+            expect(fullTrace.line.color).toEqual('#444');
+        });
+
+        it('\'colorscale\' should assume a default value if the \'color\' array is specified', function() {
+            var fullTrace = _supply({
+                line: {
+                    color: [35, 63, 21, 42]
+                },
+                dimensions: [
+                    {values: [321, 534, 542, 674]},
+                    {values: [562, 124, 942, 189]},
+                    {values: [287, 183, 385, 884]},
+                    {values: [113, 489, 731, 454]}
+                ]
+            });
+            expect(fullTrace.line).toEqual({
+                color: [35, 63, 21, 42],
+                colorscale: attributes.line.colorscale.dflt,
+                cauto: true,
+                autocolorscale: false,
+                reversescale: false,
+                showscale: false
+            });
+        });
+
+        it('\'domain\' specification should have a default', function() {
+            var fullTrace = _supply({});
+            expect(fullTrace.domain).toEqual({x: [0, 1], y: [0, 1]});
+        });
+
+        it('\'dimension\' specification should have a default of an empty array', function() {
+            var fullTrace = _supply({});
+            expect(fullTrace.dimensions).toEqual([]);
```

Comment:
What is this? A copy from `parcoords`?

Reply:
Yes, vestigial. I'll read through the code before the next round.